tools/decomp/Decomp.cpp

```diff
@@ -126,22 +126,27 @@ static bool GeneratePseudocode(llvm::Module& module,
   fin_simplifier->SetZ3Simplifier(
       // Simplify boolean structure with AIGs
       z3::tactic(fin_simplifier->GetZ3Context(), "aig") &
-      // Solve equations
-      z3::tactic(fin_simplifier->GetZ3Context(), "solve-eqs") &
+      // Cheap simplification
+      z3::tactic(fin_simplifier->GetZ3Context(), "simplify") &
       // Propagate bounds over bit-vectors
       z3::tactic(fin_simplifier->GetZ3Context(), "propagate-bv-bounds") &
       // Contextual simplification
       z3::tactic(fin_simplifier->GetZ3Context(), "ctx-simplify"));
 
-  llvm::legacy::PassManager fin;
+  llvm::legacy::PassManager scope_folder;
```

Comment:
Maybe move this declaration up above the `auto fin_simplifier{...`, and then make the creation of `fin_simplifier` conditional on `FLAGS_disable_z3`, so that we don't leak the `Z3CondSimplify` when Z3 isn't used.